app/locale/en_US/template/email/account_new.html

```diff
@@ -22,7 +22,6 @@
             <p class="highlighted-text">
                 Use the following values when prompted to log in:<br/>
                 <strong>Email</strong>: {{var customer.email}}<br/>
-                <strong>Password</strong>: {{htmlescape var=$customer.password}}
```

Comment:
What about

`<strong>Password</strong>: Password you set when creating account`